runtime/compiler/control/HookedByTheJit.cpp

```diff
@@ -6004,6 +6023,52 @@ static void suspendSamplerThreadForCheckpoint(J9VMThread *samplerThread, J9JITCo
       }
    compInfo->releaseCompMonitor(samplerThread);
    }
+
+static void triggerRecompilationForPreCheckpointGeneratedFSDBodies(J9VMThread *samplerThread, TR_J9VMBase *fe, TR::CompilationInfo *compInfo)
+   {
+   compInfo->acquireCompMonitor(samplerThread);
+
+   J9Method *method;
+   while ((method = compInfo->popForcedRecompilation()))
+      {
+      J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
+      if (
+            compInfo->isCompiled(method)
+#if !defined(J9VM_OPT_OPENJDK_METHODHANDLE)
```

Comment:
Could you please add a comment why these methods are not subject to recompilation?

Reply:
When `J9VM_OPT_OPENJDK_METHODHANDLE` is not defined, `J9AccMethodFrameIteratorSkip` is used to indicate the method is a Thunk Archetype. When `J9VM_OPT_OPENJDK_METHODHANDLE` is defined, `J9AccMethodFrameIteratorSkip` is used to indicate the method is a generated method I think.

To your second question, you're right, I don't think we need this check, because if we managed to compile the method once, we can compile it again.